### PR TITLE
some thoughts on nD data models

### PR DESCRIPTION
**Pull request by alisterburt**
I've put together some thoughts on a potential nD data model in which the `navigable world` is explicitly separated from the `rendered world` or `scene` which must be 2D or 3D. If your data is higher dimensionality than the scene then you start adding 'navigable world dimensions' which can be navigated with sliders. There's also an explicit mechanism for both dynamic projections on the CPU and for handling potentially RGB(A) data

Please let me know what you think and definitely ask any clarifying questions - this is a bit of a departure, I purposefuly haven't thought specifically about how the 'slicing' should be done but I like being able to think about the data model separately from that :)

### CODE REVIEW

File: notes/dims_slicing_thoughts.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+### nD Data
+this is arbitrary nD array data
+`data: ArrayLike[...]`
+
+This data will be transformed into one of the following for slicing
+- `dense_data: ArrayLike['... rgb(a) proj d h w']`
+- `coordinate_data: ArrayLike['... n coords']`
+
+### Renderable Data
```
Comment (tlambert03) [inline notes/dims_slicing_thoughts.md:22]: i can intuit this, but add one-sentence summary of what "renderable" data is?

File: notes/dims_slicing_thoughts.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+## Data Model
+We need to be explicit about different kinds of data, our full nD data and 
+the visualisable 2D/3D data which will be rendered at any given moment
+
+### nD Data
+this is arbitrary nD array data
+`data: ArrayLike[...]`
+
+This data will be transformed into one of the following for slicing
```
Comment (tlambert03) [inline notes/dims_slicing_thoughts.md:18]: "for" slicing? or "by" slicing?

maybe a really rudimentary diagram here?   Something like this (though, I suspect this **wrong**)

```
nD Data --[slicing]-->  ...  --[reduction]--> Renderable Data
```

**Review by tlambert03 — APPROVED** [review]
I _think_ I love it? :joy: ... there are still many gaps in my picture here, but I don't see anything that I don't like, and everything I do understand feels good.  

Perhaps some very basic definitions for dummies at the top for the terms "renderable", "navigable"

feel free to just merge :)

**Comment by alisterburt** [comment]
Thanks for taking a look and the positivity! comments were useful for my own thinking, have hopefully cleared things up a bit... merging! will start thinking about slicing

**Pull request merged**